build.rs

```diff
@@ -1,6 +1,65 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+use std::process::Command;
+use std::path::Path;
+
 fn main() {
     built::write_built_file().expect("Failed to acquire build-time information");
+    
+    // Build the frontend if it exists and Node.js is available
+    let frontend_dir = Path::new("frontend");
+    if frontend_dir.exists() {
+        println!("cargo:rerun-if-changed=frontend/src");
+        println!("cargo:rerun-if-changed=frontend/package.json");
+        
+        // Check if Node.js is available
+        if Command::new("node").arg("--version").output().is_ok() {
```

Comment:
I don't _love_ having node be required, but it does make it nice that it happens in a single build step. `node` should be added to the flake.nix dependencies (both for builds and for the dev shell) if we're going to do this